### ISSUE TITLE
Metrics on localhost:8080 from Docker image are incomplete

### ISSUE THREAD

**Issue by handelaar2** [prysmaticlabs/prysm#4326]
I try to scrape localhost:8080 for metrics to feed into Prometheus/Grafana.
The list of metrics generated from a docker image (latest on Dec19=HEAD-8c28d1) is incomplete.

When running the following command I get **77** metrics.
curl -s localhost:8080/metrics|egrep -v '^(#|bolt)' | sed -e 's/ .*//'|wc -l
(thanks @mcdee)
Doing the same for a Bazel build (cloned less than a week ago) it gives: **314**

Attached are 2 files with the individual metrics that is generated for each(bolt metrics are excluded).
[Docker_localhost-8080:metrics.txt](https://github.com/prysmaticlabs/prysm/files/3983468/Docker_localhost-8080.metrics.txt)
[Bazel_localhost-8080:metrics.txt](https://github.com/prysmaticlabs/prysm/files/3983469/Bazel_localhost-8080.metrics.txt)

The few metrics that are available (I think) come from:
cache/attestation_data.go
cache/checkpoint_state.go
cache/committee.go
state/skip_slot_cache.go

go_collector.go
go-grpc-prometheus/client_metrics.go
logrus_collector.go; process_collector.go
promhttp/http.go

So basically some Prysm cache data and the generic collectors (go;process;http;logrus;grpc)

Why does the docker image only gives a subset of metrics compared to the bazel build?

**Comment by prestonvanloon** [comment]
How can we reproduce this? 

We run the docker images in production and see all of the metrics expected metrics.

**Comment by handelaar2** [comment]
yes, Nishant has seen the same issue. 

I run:
docker pull gcr.io/prysmaticlabs/prysm/beacon-chain:latest

docker run -it -v $HOME/prysm-data:/data -p 4000:4000 -p 4001:4001 -p 13000:13000 gcr.io/prysmaticlabs/prysm/beacon-chain:latest   --datadir=/data    --initial-sync-verify-all-signatures --grpc-gateway-port 4001 --verbosity info --archive --enable-upnp --p2p-max-peers 20

Then look at the results on http://localhost:8080/metrics

**Comment by prestonvanloon** [comment]
This is normal behavior. Metrics may not appear until they have been used, especially metrics with labels.

**Comment by handelaar2** [comment]
It was not normal behaviour. The reason is that port 8080 is not forwarded in docker. As a result, it does only work partially and only some metrics are shown. Port forwarding needs to be added to docker for the metrics to work fully: **-p 8080:8080**

Issue can be closed
